src/components/WelcomePage.tsx

```diff
@@ -76,17 +79,24 @@ class WelcomePage extends React.Component<Props, any> {
   /**
    * Component did mount life-sycle event
    */
-  componentDidMount() {
+  componentDidMount = async () => {
+    const { onFacilityUpdate, onLogin } = this.props;
     const kcConf = {
       "realm": process.env.REACT_APP_KEYCLOAK_REALM as string,
       "url": process.env.REACT_APP_AUTH_SERVER_URL as string,
       "clientId": process.env.REACT_APP_AUTH_RESOURCE as string
     };
     const keycloak = new Keycloak(kcConf);
-    keycloak.init({onLoad: "login-required", checkLoginIframe: false}).success((authenticated) => {
-      this.props.onLogin && this.props.onLogin(keycloak, authenticated);
+    await keycloak.init({onLoad: "login-required", checkLoginIframe: false}).success((authenticated) => {
```

Comment:
Interesting that this was working before without being async and awaited.

Reply:
@Jdallos .success() is a legacy method for handling promises, similar to .then() and .catch() etc.

Reply:
Ok, so .success() is deprecated, could we change this functionality to remove this?

Reply:
It can be done in the future but I think in this issue, it would be out of scope.